### PR TITLE
hw: add specter wallet

### LINKED ISSUE #213: hw specter: usb support

### PR DESCRIPTION
**Pull request by edouardparis**
- [x] simulator support via tcp stream

close #213

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     steps:
       - name: Install needed libraries
         if: matrix.os == 'ubuntu-latest'
-        run: sudo apt install libexpat1-dev libxcb-composite0-dev
+        run: sudo apt install libudev-dev libexpat1-dev libxcb-composite0-dev
```
Comment (darosior) [inline .github/workflows/main.yml:22]: Wonder how it'd play out under Windows

File: src/hw/mod.rs [hunk 2]
```diff
@@ -2,21 +2,46 @@ use bitcoin::util::{bip32::DerivationPath, psbt::PartiallySignedTransaction as P
...
+    DummySigner(DummySigner),
+    SpecterSimulator(Specter<TcpStream>),
+    Specter(Specter<SerialStream>),
 }
 
 impl Channel {
     pub async fn try_connect() -> Result<Channel, Error> {
-        let device = DummySigner::try_connect("0.0.0.0:8080").await?;
-        Ok(Channel { device })
+        if let Ok(device) = DummySigner::try_connect("0.0.0.0:8080").await {
```
Comment (darosior) [inline src/hw/mod.rs:20]: Not related to this PR: the dummy signer shouldn't bind on 0.0.0.0

**Review by darosior — COMMENTED** [review]
Cool!

I am concerned that there is not a single unit test in the `hw` submodule. Also, did you look into how we could have automated testing like Stepan does with the simulator?

**Review by danielabrozzoni — APPROVED** [review]
ACK 0a60dd70d6b5047269c763acced29080a1784ec2 - I didn't review in deep the code, but I acked, delegated, spent and revaulted a vault.

**Pull request merged**